core/vision/ar_tag_detector.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+from typing import List
+from algorithms.AR_tag import Tag
+import core
+import algorithms
+
+# Dict to hold the obstacle info
+ar_tag = {}
+
+
+async def async_ar_tag_detector():
+    """
+    Async function to find obstacles.
+    """
+    while True:
+        reg_img = core.vision.camera_handler.grab_regular()
+
+        tags, reg_img = algorithms.AR_tag.detect_ar_tag(reg_img)
+
+        core.vision.feed_handler.handle_frame("artag", reg_img)
+
+        if len(tags) > 0:
+            ar_tag["detected"] = True
+            ar_tag["tags"] = tags
+        else:
+            ar_tag["detected"] = False
+            ar_tag["tags"] = []
+
+        await asyncio.sleep(1 / 30)
+
+
+def is_ar_tag():
+    """
+    Returns whether there is an obstacle.
+
+    Returns:
+    -------------
+        detect (bool) - whether or not something was detected
+    """
+    return ar_tag["detected"]
```

Comment:
Why do we need a dictionary and a Boolean to track if a artag has been detected? An empty list will evaluate False while a list with items in it will evaluate True, or we could return len(ar_tag["tags"]) - I understand the desire to have a function that returns strictly a bool, but still we could return bool(ar_tags["tags"]) or len(ar_tags["tags"]) > 0. This would allow us to completely eliminate the ar_tags dictionary and it could just be a list

Reply:
This was mostly a copy from something I did for obstacle detection, I agree and I'll change it.